appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,331 @@
+in the template contextualization, the MinIO Environment Variable File will use the defaults defined [here](https://min.io/docs/minio/linux/operations/install-deploy-manage/deploy-minio-single-node-single-drive.html#create-the-environment-variable-file).
+
+This appliance can deploy a Single-Node Single-Drive or a Single-Node Multi-Drive installation of MinIO. The configuration script
+will automatically detect how many drives are present on the VM and partition, format and mount all detected disks.
+If just one disk is detected, the appliance will configure MinIO as Single-Node Multi-Drive. If more than one additional disk are detected, 
+the appliance will configure as Single-Node Multi-Drive. 
+
+Notes for "Single-Node Multi-Drive":
+  - All drives have to be the same size since MinIO limits the size used per drive to the smallest drive in the deployment. 
+  Refer to MinIO documentation for additional information.
+  - The appliance will use /dev/vdb as the first disk for MinIO storage, and continue with vdc, vdd, etc. If any disk has a XFS partition already created, 
+the configure script will use that partition to mount on the corresponding path.
+
+After the VM has booted and app
+EOF
+)
+ONE_SERVICE_RECONFIGURABLE=true
+
+### Contextualization defaults #######################################
+
+ONEAPP_MINIO_ROOT_USER="${ONEAPP_MINIO_ROOT_USER:-myminioadmin}"
+ONEAPP_MINIO_ROOT_PASSWORD="${ONEAPP_MINIO_ROOT_PASSWORD:-minio-secret-key-change-me}"
+ONEAPP_MINIO_OPTS="${ONEAPP_MINIO_OPTS:---console-address :9001}"
+
+### Globals ##########################################################
+
+###############################################################################
+###############################################################################
+###############################################################################
+
+#
+# service implementation
+#
+
+service_cleanup()
+{
+    :
+}
+
+service_install()
+{
+    # ensuring that the setup directory exists
+    #TODO: move to service
+    mkdir -p "$ONE_SERVICE_SETUP_DIR"
+
+    # minio install
+    install_minio_server
+    # create user and group for MinIO
+    create_user_group
+    # create MinIO environment variable file
+    create_default_environment_variable_file
+    # create MinIO systemd service file
+    create_enable_systemd_service
+    # add MinIO section to fstab
+    msg info "Add MinIO section to fstab"
+    cat >> /etc/fstab <<EOF
+# Start MinIO-volumes
+# End MinIO-volumes
+EOF
+    
+    # service metadata
+    create_one_service_metadata
+
+    # cleanup
+    postinstall_cleanup
+
+    msg info "INSTALLATION FINISHED"
+
+    return 0
+}
+
+service_configure()
+{
+    # For each drive other than /dev/vda, check if they're already formatted. 
+    # If there's no format, create partitions and filesystem
+    msg info "Checking drives partition and format"
+    minio_drives=$(lsblk -d -o NAME | grep vd | grep -v 'vda')
```

Comment:
When working inside functions is good to use `local`, like so:

```suggestion
    local minio_drives=$(lsblk -d -o NAME | grep vd | grep -v 'vda')
```